analytical_engine/core/context/java_pie_projected_default_context.h

```diff
@@ -313,9 +181,66 @@ class JavaPIEProjectedDefaultContextWrapper
     VLOG(0) << "Java env not available";
     return NULL;
   }
-  std::shared_ptr<IFragmentWrapper> frag_wrapper_;
-  std::shared_ptr<context_t> ctx_;
-  std::shared_ptr<IContextWrapper> _inner_context_wrapper;
+};
+
+// shall not be invoked
+template <typename FRAG_T>
+class JavaPIEProjectedDefaultContextWrapper
+    : public IJavaPIEProjectedDefaultContextWrapper {
+  using fragment_t = FRAG_T;
+  using label_id_t = typename fragment_t::label_id_t;
+  using prop_id_t = typename fragment_t::prop_id_t;
+  using oid_t = typename fragment_t::oid_t;
+  using context_t = JavaPIEProjectedDefaultContext<fragment_t>;
+
+ public:
+  JavaPIEProjectedDefaultContextWrapper(
+      const std::string& id, std::shared_ptr<IFragmentWrapper> frag_wrapper,
+      std::shared_ptr<context_t> context) {}
+
+  std::string context_type() override {
+    return std::string(CONTEXT_TYPE_JAVA_PIE_PROJECTED_DEFAULT);
+  }
+
+  std::shared_ptr<IFragmentWrapper> fragment_wrapper() override {
+    return std::shared_ptr<IFragmentWrapper>(nullptr);
+  }
+  // Considering labeledSelector vs selector
+  bl::result<std::unique_ptr<grape::InArchive>> ToNdArray(
+      const grape::CommSpec& comm_spec, const Selector& selector,
+      const std::pair<std::string, std::string>& range) override {
+    return std::make_unique<grape::InArchive>();
```

Comment:
use `RETURN_GS_ERROR(ErrorCode::kUnimplementedMethod, "No need to implement for java context wrapper")` instead.

Reply:
Done.